lib/SPIRV/libSPIRV/SPIRVValue.h

```diff
@@ -98,8 +98,36 @@ class SPIRVValue : public SPIRVEntry {
 
   void setAlignment(SPIRVWord);
   void setVolatile(bool IsVolatile);
-  void setNoSignedWrap(bool HasNoSignedWrap);
-  void setNoUnsignedWrap(bool HasNoUnsignedWrap);
+
+  template <spv::Decoration NoIntegerWrapDecoration>
```

Comment:
Just a nit: template function definition may be left in cpp file. One can just write explicit function specializations after the definition to eliminate linkage errors.

Reply:
Oh, I was trying to do that, but failed, but I just realized, that I tried it within a class, but should do specialization outside of the class definition.